src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -41,17 +39,27 @@ public void AddLocalSuppression (CustomAttribute ca, ICustomAttributeProvider pr
 
 		public bool IsSuppressed (int id, MessageOrigin warningOrigin, out SuppressMessageInfo info)
 		{
-			if (HasLocalSuppressions && warningOrigin.MemberDefinition != null) {
-				IMemberDefinition memberDefinition = warningOrigin.MemberDefinition;
-				while (memberDefinition != null) {
-					if (IsLocallySuppressed (id, memberDefinition, out info))
-						return true;
+			info = default;
+			if (warningOrigin.MemberDefinition == null)
+				return false;
 
-					memberDefinition = memberDefinition.DeclaringType;
-				}
+			IMemberDefinition memberDefinition = warningOrigin.MemberDefinition;
+			if (IsGloballySuppressed (id, memberDefinition, out info))
+				return true;
+
+			// Check if there's a module level suppression.
```

Comment:
I think module level suppressions should be done after we do the while loop - in general I expect local suppressions to be on the method - so that should be the "fast path" through this function.